pkg/sql/information_schema.go

```diff
@@ -552,6 +588,25 @@ func forEachColumnInTable(
 	return nil
 }
 
+func forEachForeignKeyInTable(
+	table *sqlbase.TableDescriptor,
+	fn func(
+		*sqlbase.IndexDescriptor,
+		*sqlbase.ForeignKeyReference,
+	) error,
+) error {
+	for i := range table.Indexes {
```

Comment:
The issue is that `GetConstraintInfo` refetches all of the referenced table descriptors for each FK. That's fine if we only want to get the constraint info for a single table, but we already fetched all descriptors here, so we really want to avoid the quadratic blowup. 

If that alone was the issue, it could probably be fixed with some parameterized method to get table descriptors from an ID, but I'm also not sure that `ConstraintDetail` is structured enough for this use case. I would need to add a few new fields to the struct (like `FKColumnIDs` and `IndexColumnIDs`), most of which would be specific to only one constraint type. I would also need to add references to index and table descriptors to the structure. Does this seem like the right approach?